src/Illuminate/Cache/RedisTaggedCache.php

```diff
@@ -93,8 +93,44 @@ public function forever($key, $value)
      */
     public function flush()
     {
-        $this->flushValues();
-        $this->tags->flush();
+        /** @var \Illuminate\Cache\RedisStore $redis_store */
+        $redis_store = $this->store;
+
+        $redis_connection = $redis_store->connection();
+        $redis_prefix = match ($redis_connection::class) {
+            \Illuminate\Redis\Connections\PhpRedisConnection::class => $redis_connection->client()->getOption(\Redis::OPT_PREFIX),
+            \Illuminate\Redis\Connections\PredisConnection::class => $redis_connection->client()->getOptions()->prefix,
+        };
+
+        /** @var \Illuminate\Cache\RedisTagSet $redis_tag_set */
+        $redis_tag_set = $this->tags;
+        $entries = $redis_tag_set->entries();
+
+        $cache_tags = [];
+        $need_deleted_keys = [];
+
+        $cache_prefix = $redis_prefix.$redis_store->getPrefix();
+
+        foreach ($this->tags->getNames() as $name) {
+            $cache_tags[] = $cache_prefix.$this->tags->tagId($name);
+        }
+
+        foreach ($entries as $entry) {
+            $need_deleted_keys[] = $redis_store->getPrefix().$entry;
+        }
+
+        $lua_script = <<<'LUA'
+            local prefix = table.remove(ARGV, 1)
+            for i, key in ipairs(KEYS) do
+                redis.call('DEL', key)
+                for j, arg in ipairs(ARGV) do
+                    local zkey = string.gsub(key, prefix, "")
+                    redis.call('ZREM', arg, zkey)
```

Comment:
I believe this line violates:

> Important: to ensure the correct execution of scripts, both in standalone and clustered deployments, all names of keys that a script accesses must be explicitly provided as input key arguments. The script should only access keys whose names are given as input arguments. Scripts should never access keys with programmatically-generated names or based on the contents of data structures stored in the database.

Because the key name is taken from `ARGV` instead of `KEYS`.

Reply:
The `KEYS` and `ARGV` are not the same.
`KEYS` represent cache entries within the cache tag, while `ARGV` stands for cache tag keys.
I utilize key for KEYS and zkey for ARGV.

Reply:
The `arg` parameter is the key that `ZREM` operates on and it is taken from `ARGV`.